src/components/Select/Select.tsx

```diff
@@ -81,6 +81,17 @@ const Select = React.forwardRef<HTMLInputElement, SelectProps>((props, ref) => {
         onBackspace: () => {
           debouncedOnChange('');
           setIsOpen(true);
+
+          /** When the textField's value equals the selectedOption's label we should clear the field*/
+          if (!isMulti) {
+            if (selectedOption && onChange) {
+              if (selectedOption.label === textFieldValue) {
```

Comment:
why 3 if here, just wondering?

Reply:
mostly for readability reasons, although in the future this component will be refactored to reduce Cognitive Complexity